pkg/controller/controller_utils/controller_utils.go

```diff
@@ -0,0 +1,53 @@
+package controller_utils
+
+import (
+	"strings"
+	"unicode"
+)
+
+const (
+	metricsNameDelimiter = "_"
+	loggerNameDelimiter  = "-"
+)
+
+type ControllerNamer interface {
+	MetricsName() string
+	LoggerName() string
+}
+
+type ControllerName []string
```

Comment:
we shouldn't expose this `controllerName`. Doesn't make sense to expose concrete type and interface because it makes the recommended usage not obvious.

Reply:
agreed